doc/MiniLanguageTutorial.md

```diff
@@ -480,37 +563,278 @@ Mini never automatically converts types to make an operation succeed.  Programme
+   A *KvsNode* may be one of 3 types, it may be an integer of value `0`, a `Tuple` of length 2, or a `Tuple` of length 8.
+   In the case of a 2 `Tuple`, the first value in the tuple must be valid for *keytype*, and the second must be valid for `option<`*valuetype*`>`.
+   In the case of an 8 tuple, each value in the tuple must be a valid *KvsNode*.
+1. string:  
+   As a string is equivalent to `(uint, buffer)`, the valid AVM values are those values that are valid for `(uint, buffer)`,
+   in other words all values must be a `Tuple(Int, Buffer)`. Any possible value for this representation is allowable,
+   although strings are always initialized such that `Int` represents the length of the `Buffer`. 
+1. any:  
+   Any AVM value may be used for any, the compiler makes no assumptions about the value of this type.
+1. option<*type*>:  
+   Either a length 1 tuple containing `0`, or a length 2 tuple where the first field is `1` and the second field is a valid value for *type*.
+1. every:  
+   There are no valid AVM values for this type, the compiler assumes it will never exist.
+1. union<*type1*, *type2*, ... , *typeN*>:  
+   The valid AVM values for this type are the union of valid values for *type1*, *type2*, ... , *typeN*. In other words, an AVM value is valid for this type if and only if there is some type in *type1*, *type2*, ... , *typeN* for which the value is valid.
+1. func:  
+   All function types are represented internally as a `CodePoint` pointing to the start of the function in memory.
+1. nominal/generic:  
+   A nominal type has the same AVM representation as its representation.
+   The representation of a nominal type is the type on the right half of the type declaration.
+   This also applies to specialized generic types, where the AVM representation is equivalent to the type resulting from
+   replacing each type variable with the concrete type specified in the specialization.
+
+## Wrapping tuple
+
+Mini uses the concept of a wrapping tuple to express tuples and structs of length greater than 8. These are expressed by nesting `Tuple`s
+inside of other types, in order to allow more "slots" to fit in a single value.
+The structure of a wrapping tuple is a single `Tuple` from length 0 to 8.
+If the length is 9 or greater, the base of the wrapping tuple of length *length*, is a nested `Tuple` of uniform depth *N*.
+Where *N*, is the greatest integer such that `8^`*N* `<=` *length*. A nested tuple of depth 1 is a single length 8 `Tuple`,
+and for all greater *N* it is a length 8 `Tuple` whose fields are all depth *N*`-1` tuples.
+
+The order of the fields in a nested `Tuple` is the left to right order in a depth 1 tuple, and in a length *N* tuple,
+it is the order of the fields in each depth *N*`-1`, concatenated in left to right order.
+
+A wrapping tuple of length *length* is a depth *N* nested tuple, where *N* is the greatest integer that `8^`*N*`<=`*length*,
```

Comment:
This is looks like a separate, inconsistent description of of wrapping tuples.   Need to eliminate one of the two.  Also, in this one, need to say how the tuple contents relate to the underlying type.

Reply:
The above definition describes "nested tuples", which was introduced in order to aid the description of wrapping tuples. 

I recognize the existence of two definitions can be confusing, however I cannot think of an alternative way to describe wrapping tuples that doesn't either partly rely on intuition or isn't overall more confusing.